source/math/FloatRandomGenerator.ooc

```diff
@@ -27,22 +27,30 @@ FloatRandomGenerator: abstract class {
 }
 
 FloatUniformRandomGenerator: class extends FloatRandomGenerator {
-	_state: Int
-	_min := 0.0f
-	_max := 1.0f
+	_state: UInt
+	_min, _max, _rangeCoefficient: Float
+	minimum ::= this _min
+	maximum ::= this _max
 	init: func {
 		this _state = Time microtime()
+		this updateRange(0.0f, 1.0f)
 	}
 	init: func ~withSeed (seed: Int) {
 		this _state = seed
+		this updateRange(0.0f, 1.0f)
 	}
-	init: func ~withParameters (=_min, =_max, seed := Time microtime()) {
+	init: func ~withParameters (min, max: Float, seed := Time microtime()) {
 		this _state = seed
+		this updateRange(min, max)
+	}
+	updateRange: func (=_min, =_max) {
+		this _rangeCoefficient = (1.0f / 4294967295.0f) * (this _max - this _min)
```

Comment:
To the uninitiated reader, it would be nice to have some form of explanation for this magic number.